src/api/exchange/DepositApiMock.ts

```diff
@@ -107,15 +107,15 @@ export class DepositApiMock implements DepositApi {
     await this._erc20Api.transfer(tokenAddress, userAddress, this.getContractAddress(), amount)
 
     log(`[DepositApiMock] Deposited ${formatAmount(amount)} for token ${tokenAddress}. User ${userAddress}`)
-    return { data: undefined, receipt: RECEIPT }
+    return RECEIPT
   }
 
   public async requestWithdraw(
     userAddress: string,
     tokenAddress: string,
     amount: BN,
     txOptionalParams?: TxOptionalParams,
-  ): Promise<TxResult<void>> {
+  ): Promise<Receipt> {
```

Comment:
This change is good, BUT it's not part of this PR. I think in this situations is more clear to create on PR with this change and the refactor of all the app. Then this #140 would use that PR's branch as it's base

Please, keep this in mind so PRs are easier to review.

Reply:
```
real_wallet PR 
                            \
                              \
                                   this PR
                             /
                            /
type change PR
```
I chose to avoid diamond problem with inheritance. also avoid a mid-step in the waterfall

Reply:
I think I would have done a waterfall here.
`real_wallet PR` --> `type change PR` --> `this PR`

or if you prefer:
`type change PR` --> `real_wallet PR`  --> `this PR`

We are all trying to do this to make PRs easier to review for others. 
For this one is fine now, wasn't that bad, but AGAIN, we need to keep the scope of a PR into one single thing please for next PRs. Don't need to take it to an extreme, but just something that makes sense.

Now we are more in the project, so difficulting the review has a multiplying effect. Even for the one making the PR. And I mean not just people in front end, I think it's important for people working in other parts of dfusion.
 
Thanks for understanding!

Reply:
Sure, shoudl have gone with either of those. Hindsight is 20/20 :stuck_out_tongue_closed_eyes:

Reply:
Also, it would have been good to invest in bitcoin in 2009

Reply:
> Also, it would have been good to invest in bitcoin in 2009

The list of emoji reactions don't include a :cry: nor a :sob: